scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -828,6 +830,44 @@
       (log/error t "In" pool-name "pool, error launching tasks for"
                  (cc/compute-cluster-name compute-cluster) "compute cluster"))))
 
+(defn filter-matches-for-ratelimit
+  "Given a set of matches, determine which compute clusters are beyond the rate limit and filter matches in those compute clusters out."
+  [matches]
+  (let [augmented-matches (->> matches
+                               (group-by match->compute-cluster)
+                               (map
+                                 (fn [[compute-cluster matches-in-compute-cluster]]
+                                   (let [compute-cluster (or compute-cluster )
+                                         compute-cluster-name (if compute-cluster
+                                                                (cc/compute-cluster-name compute-cluster)
+                                                                ; In unit tests we may not have a compute cluster.
+                                                                ; If so, swap this compute cluster in.
+                                                                "fake-compute-cluster-name")
```

Comment:
Should we just fix the unit tests? This doesn't smell great to me

Reply:
I can try a bit further. There's  (unfortunately) a lot of infrastructure I'd have to build, which isn't desirable either.

Reply:
Aaah. That was true with an older version of the patch where I was integrating at the wrong location. Unnecessary now!